Text/PrettyPrint/Compact/Core.hs

```diff
@@ -88,46 +122,78 @@ pareto' acc (x:xs) = if any (≺ x) acc
                             -- for all y ∈ acc, y <= x, and thus x ≺ y
                             -- is false. No need to refilter acc.
 
+leFst :: Ord a => (a, b) -> (a, b) -> Bool
+leFst = (<=) `on` fst
 
```

Comment:
What is the purpose of removing the Poset class? Please try to stick to one atomic change per PR.